crates/tick/src/delay.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::pin::Pin;
+use std::task::{Context, Poll, Waker};
+use std::time::Duration;
+
+use super::Clock;
+use super::timers::TimerKey;
+
+/// Asynchronously delays for the specified duration.
+///
+/// # Precision
+///
+/// The delay uses the current thread's scheduler to schedule its ticks. The precision
+/// of the delay is affected by the load on this thread. There are no guarantees about the
+/// precision of the delay other than that it will eventually complete. When the thread is healthy,
+/// the delay should be close to the specified duration.
+///
+/// Note: `Delay` is not affected by changes in the system clock.
+///
+/// # Examples
+///
+/// ```
+/// use std::time::Duration;
+///
+/// use tick::{Clock, Delay};
+///
+/// # async fn delay_example(clock: &Clock) {
+/// let stopwatch = clock.stopwatch();
+///
+/// // Delay for 10 milliseconds
+/// clock.delay(Duration::from_millis(10)).await;
+///
+/// assert!(stopwatch.elapsed() >= Duration::from_millis(10));
+/// # }
+/// ```
+#[derive(Debug)]
+pub struct Delay {
+    // Currently scheduled timer. This value is not initialized before
```

Comment:
Does this mean that the delay until starts counting the first time the application calls await on something?